src/combined-selectors/reporting-config-search-results.ts

```diff
@@ -54,16 +55,100 @@ function searchReportingConfig(
 		}
 	}
 
+	// Finally, search for the search term in the description of the products and feature areas
+	const searchTermTokens = tokenizeAndNormalize( searchTerm );
+	const scores: Record< string, number > = {};
+
+	for ( const token of searchTermTokens ) {
+		const matchingEntities = invertedIndex.get( token ) || [];
+
+		for ( const { id, weight } of matchingEntities ) {
+			if ( ! scores[ id ] ) {
+				scores[ id ] = 0;
+			}
+			scores[ id ] += weight;
+		}
+	}
+
+	const scoreThreshold = 1;
+
+	for ( const entityId in scores ) {
+		if ( scores[ entityId ] >= scoreThreshold ) {
+			if ( products[ entityId ] ) {
+				searchResults.products.add( entityId );
+			} else if ( featureGroups[ entityId ] ) {
+				searchResults.featureGroups.add( entityId );
+				searchResults.products.add( featureGroups[ entityId ].productId );
+			} else if ( features[ entityId ] ) {
+				searchResults.features.add( entityId );
+				if ( features[ entityId ].parentType === 'product' ) {
+					searchResults.products.add( features[ entityId ].parentId );
+				} else {
+					searchResults.featureGroups.add( features[ entityId ].parentId );
+					const parentFeatureGroup = featureGroups[ features[ entityId ].parentId ];
+					searchResults.products.add( parentFeatureGroup.productId );
+				}
+			}
+		}
+	}
+
 	return searchResults;
 }
 
+function addTokensToInvertedIndex(
+	entityId: string,
+	description: string | undefined,
+	invertedIndex: Map< string, Array< { id: string; weight: number } > >
+) {
+	const descriptionTokens = description ? tokenizeAndNormalize( description ) : [];
+	const tokensWithWeights = [ ...descriptionTokens.map( ( token ) => ( { token, weight: 1 } ) ) ];
```

Comment:
**Question:** Just for my understanding, is the weight here to add flexibility in case we wanted to weight certain words more in the future? Were there some enhancement ideas you had in mind?

Reply:
Good question! I originally implemented this function for names, keywords, and descriptions. We can take the weight out.